connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -639,4 +639,12 @@ abstract class Catalog {
    * @since 3.5.0
    */
   def listCatalogs(): Dataset[CatalogMetadata]
+
+  /**
+   * Returns a list of catalogs which name match the specify pattern and available in this
+   * session.
+   *
+   * @since 3.5.0
+   */
+  def listCatalogs(pattern: String): Dataset[CatalogMetadata]
```

Comment:
If we accept such API, do we prefer to use existing `def listCatalogs()` and have `def listCatalogs(pattern: Option[String])`? cc @cloud-fan

Reply:
If we can, I think it's OK too.

Reply:
> If we accept such API, do we prefer to use existing `def listCatalogs()` and change it to `def listCatalogs(pattern: Option[String])`?

it will break the binary compatiablity

Reply:
> > If we accept such API, do we prefer to use existing `def listCatalogs()` and change it to `def listCatalogs(pattern: Option[String])`?
> 
> it will break the binary compatiablity

ah good point.

Reply:
@zhengruifeng in fact, what kind of binary compatibility. 

The suggested change is still compatibility with existing code? The only way I can think that will break is through reflection (so tried to load `listCatalogs()` but there is no such signature). Is this the binary compatibility referred here?

Reply:
I meant that if we change `def listCatalogs()` to `def listCatalogs(pattern: Option[String])`, it breaks the binary